### PR TITLE
feat: fix logic and refactor app install

### PR DESCRIPTION
**Pull request by chrisduma-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - ...

### 📝 Description

We had an issue where if the app was not installed the sell/card would crash. This PR fixes it.
Also, i have refactored the logic a little.

### ❓ Context

[Ticket](https://ledgerhq.atlassian.net/browse/LIVE-11524)

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by CremaFR — DISMISSED** [review]
I've done the QA for swaps no issue ;)

**Review by sarneijim — APPROVED** [review]
We need applied the same change for FUND.
If we can merge today, tomorrow you need to deliver a new exchangeSDK version (patch) and send it to Mercuryo and Moonpay.

File: libs/ledger-live-common/src/hw/actions/startExchange.ts [hunk 1]
```diff
@@ -129,38 +129,20 @@ export const createAction = (
...
-      if (!exchange || !mainFromAccount || !mainToAccount) {
-        return {
-          appName: "Exchange",
-          requireLatestFirmware,
-        };
-      } else {
-        const shouldAddEthApp =
-          (mainFromAccount.currency.family === "evm" || mainToAccount.currency.family === "evm") &&
-          mainFromAccount.currency.managerAppName !== "Ethereum" &&
-          mainToAccount.currency.managerAppName !== "Ethereum";
```
Comment (Justkant) [inline libs/ledger-live-common/src/hw/actions/startExchange.ts:141]: Looks like you removed a recent fix I did here, any reason we removed it ?

Comment (chrisduma-ledger) [reply libs/ledger-live-common/src/hw/actions/startExchange.ts:141]: Hello. Yes, I have removed that and just added this part:
`dependencies.push({ appName: mainFromAccount?.currency?.managerAppName });`

It makes the code easier to read, and adds any app as needed. @CremaFR tested it for swaps, and it works well.

Comment (Justkant) [reply libs/ledger-live-common/src/hw/actions/startExchange.ts:141]: Yes but we need to keep the above logic to add ethereum in case of avax as it's need for swap

Comment (CremaFR) [reply libs/ledger-live-common/src/hw/actions/startExchange.ts:141]: Could you please add a comment next time to explain such changes? This was incredibly obscure, and no one could have guessed it without context. 
Even just linking the JIRA ticket would have been enough, as we often do for similar quirks in swaps https://github.com/LedgerHQ/ledger-live/blob/develop/apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx#L388

Comment (CremaFR) [reply libs/ledger-live-common/src/hw/actions/startExchange.ts:141]: Or put in a function like `addAvaxAppDep `

Comment (CremaFR) [reply libs/ledger-live-common/src/hw/actions/startExchange.ts:141]: And I'll make a PR to add PTX owner of this file too from now on

Comment (Justkant) [reply libs/ledger-live-common/src/hw/actions/startExchange.ts:141]: Yes sure, the name of the variable and the logic was not that hard to follow IMO and I understand that you probably wanted to remove this but if something is obscure or doesn't make sense I usually find the author of the change and try to get answers before doing a change if none are provided in the source code
I'll keep this in mind if I end up in a similar situation, as I usually add context if needed for strange changes

**Review by CremaFR — DISMISSED** [review]

**Review by Justkant — DISMISSED** [review]

**Review by sarneijim — DISMISSED** [review]

**Review by jiyuzhuang — DISMISSED** [review]

**Review by sarneijim — APPROVED** [review]

**Review by Justkant — APPROVED** [review]

**Review by CremaFR — APPROVED** [review]
